File: CF/362/B/B1.cpp
/**
 * File              : B1.cpp
 * Author            : Bao To Hoai
 * Date              : 02.11.2020 14:29:22 UTC+7
 * Last Modified Date: 02.11.2020 14:40:20 UTC+7
 * Last Modified By  : Bao To Hoai
 */
#include <bits/stdc++.h>
using namespace std;

#define ll long long
#define all(c) (c).begin(), (c).end()
#define sz(x) (int)(x).size()

string to_string(char c) {
    return string(1, c);
}

string to_string(string s) {
    return s;
}

template<class A, class B> string to_string(pair<A, B> p) {
    return "(" + to_string(p.first) + ", " + to_string(p.second) + ")";
}

template<class T> string to_string(T v) {
    bool f = true;
    string res = "{";
    for (auto &x : v) {
        if (!f) {
            res += ", ";
        }
        f = false;
        res += to_string(x);
    }
    res += "}";
    return res;
}

void DBG() {
    cerr << "]" << endl;
}

template<class H, class... T> void DBG(H h, T... t) {
    cerr << to_string(h);
    if(sizeof...(t))
        cerr << ", ";
    DBG(t...);
}

#ifdef LOCAL
#define dbg(...) cerr << "LINE(" << __LINE__ << ") -> [" << #__VA_ARGS__ << "]: [", DBG(__VA_ARGS__)
#else
#define dbg(...) 42
#endif

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(0);
    int n, m;
    cin >> n >> m;
    if (m == 0) {
        puts("YES");
        return 0;
    }
    vector<int> A(m);
    for (int i = 0; i < m; i++) {
        cin >> A[i];
    }
    sort(all(A));
    if (A[0] == 1 || A[m - 1] == n) {
        puts("NO");
        return 0;
    }
    for (int i = 0; i < m - 2; i++) {
        if (A[i] + 1 == A[i + 1] && A[i] + 2 == A[i + 2]) {
            puts("NO");
            return 0;
        }
    }
    puts("YES");

    return 0;
}

File: CF/362/B/B1.java
/**
 * File              : B1.java
 * Author            : Bao To Hoai
 * Date              : 02.11.2020 14:40:56 UTC+7
 * Last Modified Date: 02.11.2020 14:45:58 UTC+7
 * Last Modified By  : Bao To Hoai
 */
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;
import java.util.Arrays;

public class B1 {

    public static void main(String[] args) {
        FastScanner fs = new FastScanner();
        int n = fs.nextInt();
        int m = fs.nextInt();
        if (m == 0) {
            System.out.println("YES");
            return;
        }
        int[] A = new int[m];
        for (int i = 0; i < m; i++) {
            A[i] = fs.nextInt();
        }
        Arrays.sort(A);
        if (A[0] == 1 || A[m - 1] == n) {
            System.out.println("NO");
            return;
        }
        for (int i = 0; i < m - 2; i++) {
            if (A[i] + 1 == A[i + 1] && A[i] + 2 == A[i + 2]) {
                System.out.println("NO");
                return;
            }
        }
        System.out.println("YES");
    }
    
    static class FastScanner {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer("");
        
        String next() {
            while (!st.hasMoreTokens())
                try {
                    st = new StringTokenizer(br.readLine());
                } catch (IOException e) {
                    e.printStackTrace();
                }
            return st.nextToken();
        }
        
        int nextInt() {
            return Integer.parseInt(next());
        }
              
        long nextLong() {
            return Long.parseLong(next());
        }
    }
}


